src/mono/mono/metadata/metadata.c

```diff
@@ -49,7 +49,7 @@ typedef struct {
 static gboolean do_mono_metadata_parse_type (MonoType *type, MonoImage *m, MonoGenericContainer *container, gboolean transient,
 					 const char *ptr, const char **rptr, MonoError *error);
 
-static gboolean do_mono_metadata_type_equal (MonoType *t1, MonoType *t2, gboolean signature_only);
+static gboolean do_mono_metadata_type_equal (MonoType *t1, MonoType *t2, gboolean signature_only, gboolean skip_cmods_check);
 static gboolean mono_metadata_class_equal (MonoClass *c1, MonoClass *c2, gboolean signature_only);
 static gboolean mono_metadata_fnptr_equal (MonoMethodSignature *s1, MonoMethodSignature *s2, gboolean signature_only);
 static gboolean _mono_metadata_generic_class_equal (const MonoGenericClass *g1, const MonoGenericClass *g2,
```

Comment:
two things:

1. Whenever we start adding a ton of boolean parameters to a function, I get a warning going off in my head.  The calls now begin to look like `do_mono_metadata_type_equal (..., TRUE, FALSE)` and that's really hard to read and easy to make mistakes.
   Add a new enum bitmask - `MONO_TYPE_EQ_FLAGS_xyz` and just pass a single enum value instead of a ton of boolean arguments
2. I think `mono_metadata_fnptr_equal` needs to have the `skip_cmods_check` value propagated to it, too.  if you're checking the equivalence of two methods that take a function pointer argument, those arguments need to be recursively checked using the match/ignore cmods mode.

Reply:
Addressed #1

Reply:
Addressed #2